buildenv/jenkins/JenkinsfileBase

```diff
@@ -596,12 +598,30 @@ def uploadToArtifactory(pattern) {
 		def artifactoryUrl = server.getUrl()
 		def uploadUrl = "${artifactoryUrl}/${artifactoryUploadDir}"
 		echo "Test output artifactory URL:'${uploadUrl}'"
-		currentBuild.description += "<br><a href=${uploadUrl}>Artifacts</a>"
+		if (!currentBuild.description.contains(uploadUrl)) {
+			currentBuild.description += "<br><a href=${uploadUrl}>Artifacts</a>"
+		}
 	} else {
 		echo "ARTIFACTORY_SERVER is not set. Artifacts are not uploaded onto artifactory server."
 	}
 }
 
+def addGrinderLink() {
+	def	customTargetOpt = params.CUSTOM_TARGET ? "&CUSTOM_TARGET=${params.CUSTOM_TARGET}" :  ""
+	def	sdkSourceOpt = params.SDK_RESOURCE ? "&SDK_RESOURCE=${params.SDK_RESOURCE}" :  ""
+	def	sdkUrlOpt = params.CUSTOMIZED_SDK_URL ? "&CUSTOMIZED_SDK_URL=${params.CUSTOMIZED_SDK_URL}" : ""
+	def	sdkUrlCredentialOpt = params.CUSTOMIZED_SDK_URL_CREDENTIAL_ID ? "&CUSTOMIZED_SDK_URL_CREDENTIAL_ID=${params.CUSTOMIZED_SDK_URL_CREDENTIAL_ID}" : ""
+
+	def url = "${HUDSON_URL}/job/Grinder/parambuild/?" \
+	+ "JDK_VERSION=${params.JDK_VERSION}" \
+	+ "&JDK_IMPL=${params.JDK_IMPL}" \
+	+ "&BUILD_LIST=${params.BUILD_LIST}" \
+	+ "&JenkinsFile=${params.JenkinsFile}" \
+	+ "&TARGET=${params.TARGET}" \
+	+ "${sdkSourceOpt}${sdkUrlOpt}${sdkUrlCredentialOpt}${customTargetOpt}"
+	currentBuild.description += "<br><a href=${url}>Rerun in Grinder</a> Note: TARGET field can be modified to run a failed test"
```

Comment:
I would drop the "Note:" part, as its a tight space, and here are some suggested minor wording options:
Or "TARGET should be changed to run only the failed test target"
Or "TARGET can be changed to run only the failed test target"
Or "TARGET can be modified to run the failed test target" , ["rather than the top-level test target"]...

Reply:
updated

![image](https://user-images.githubusercontent.com/19273206/65438305-453e1a00-ddf3-11e9-8bd1-7272e3a08fa7.png)
https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder_Advanced/610/